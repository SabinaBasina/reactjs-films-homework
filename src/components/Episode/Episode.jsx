/* eslint-disable react/no-danger */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Episode.scss';
import NoImageEpisode from './NoImageEpisode.jpg';


class Episode extends PureComponent {
  render() {
    const { data } = this.props;
    return (

      <div className={styles.Episode}>

        <b className={styles.Number}>
          {data.season}
            x
          {data.number}
        </b>

        <img src={data.image ? data.image.medium : NoImageEpisode} alt="EpisodeImage" />

        <div className={styles.About}>
          <b>
            {data.name}
          </b>
          <br />
          <p dangerouslySetInnerHTML={{ __html: data.summary }} />
        </div>

      </div>
    );
  }
}

export default Episode;

Episode.propTypes = {
  data: PropTypes.objectOf(PropTypes.object).isRequired,
};

// Episode.defaultProps = {
//   data: 'undefined',
// };
