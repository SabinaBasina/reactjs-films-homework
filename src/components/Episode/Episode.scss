.Episode {
  display: block;
  margin: 0 20% 70px 20%;
}

.Episode::after {
  content: " ";
  display: table;
  clear: both;
}

.Number {
  float: left;
  margin-right: 30px;
}

.Episode img {
  float: left;
  margin: 0 30px 10px 0;
}

@media only screen and (max-width: 1100px) {
  .Episode {
    margin: 0 10% 100px 10%;
  }
}

@media only screen and (max-width: 820px) {
  .Episode {
    margin: 0 5% 100px 5%;
  }

  .Number {
    float: none;
    display: block;
    margin: 0 auto 15px auto;
    text-align: center;
  }

  .Episode img {
    float: none;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }
}
