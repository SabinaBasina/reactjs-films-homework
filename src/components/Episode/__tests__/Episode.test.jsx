import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Episode from '../Episode';

test('Episode test img is', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Episode data={{ image: { medium: 'medium' } }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

test('Episode test img no', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Episode data={{ }} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
