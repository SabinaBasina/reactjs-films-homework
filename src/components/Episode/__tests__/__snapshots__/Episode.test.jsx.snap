// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode test img is 1`] = `
<div
  className="Episode"
>
  <b
    className="Number"
  >
    x
  </b>
  <img
    alt="EpisodeImage"
    src="medium"
  />
  <div
    className="About"
  >
    <b />
    <br />
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Episode test img no 1`] = `
<div
  className="Episode"
>
  <b
    className="Number"
  >
    x
  </b>
  <img
    alt="EpisodeImage"
    src="test-file-stub"
  />
  <div
    className="About"
  >
    <b />
    <br />
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;
